File: Data Structures & Algorithms/string-matching-in-an-array/submission-2.py
class Solution:
    # brute-force solution, O(m^2 * n^2) time, O(1) space
    # n: number of words, m: length of longest word
    def stringMatching(self, words: List[str]) -> List[str]:
        result = set()
        for i in range(len(words)):
            for j in range(len(words)):
                if i != j and words[i] in words[j]:
                    result.add(words[i])
                    break

        return list(result)

File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-2.cpp
class Solution {
public:
    // brute-force solution, O(n^2) time, O(1) space
    int longestMonotonicSubarray(vector<int>& nums) {
        int n = nums.size();
        int longest = 1;

        for (int i = 0; i < n - 1; i++) {
            int currStreak = 1;
            for (int j = i + 1; j < n; j++) {
                if (nums[j] == nums[j - 1] || ((nums[i] < nums[i + 1]) != (nums[j - 1] < nums[j]))) {
                    break;
                }
                currStreak++;
            }
            longest = max(longest, currStreak);
        }
        return longest;
    }
};